kmk/extensions/oled.py

```diff
@@ -0,0 +1,291 @@
+import busio
+import gc
+from supervisor import ticks_ms
+
+import adafruit_displayio_ssd1306
+import displayio
+import terminalio
+from adafruit_display_text import label
+
+from kmk.extensions import Extension
+from kmk.handlers.stock import passthrough as handler_passthrough
+from kmk.keys import make_key
+from kmk.kmktime import check_deadline
+
+
+class OledEntryType:
+    TXT = 0
+    IMG = 1
+
+
+class OledData:
+    def __init__(
+        self,
+        entries=None,
+    ):
+        if entries is not None:
+            self.data = entries
+
+    @staticmethod
+    def oled_text_entry(
+        text='',
+        x=0,
+        y=0,
+        x_anchor='left',
+        y_anchor='top',
+        direction='LTR',
+        line_spacing=0.75,
+        inverted=False,
+        layer=None,
+        side=None,
+    ):
+        _x = 0
+        _x_anchor = 0.0
+        _y_anchor = 0.0
+        if x_anchor == 'left' or x_anchor == 'L':
+            _x_anchor = 0.0
+        elif x_anchor == 'middle' or x_anchor == 'M':
+            _x_anchor = 0.5
+        elif x_anchor == 'right' or x_anchor == 'R':
+            _x_anchor = 1.0
+        else:
+            _x_anchor = 0.0
+        if y_anchor == 'top' or y_anchor == 'T':
+            _y_anchor = 0.0
+        elif y_anchor == 'middle' or y_anchor == 'M':
+            _y_anchor = 0.5
+        elif y_anchor == 'bottom' or y_anchor == 'B':
+            _y_anchor = 1.0
+        else:
+            _y_anchor = 0.0
+        if (
+            x_anchor != 'middle'
+            or x_anchor != 'right'
+            or x_anchor != 'm'
+            or x_anchor != 'r'
+        ):
+            _x = x + 1
+        else:
+            _x = x
+        return {
+            0: text,
+            1: _x,
+            2: y,
+            3: side,
+            4: layer,
+            5: OledEntryType.TXT,
+            6: _x_anchor,
+            7: _y_anchor,
+            8: direction,
+            9: line_spacing,
+            10: inverted,
+        }
```

Comment:
Why is this a dictionary? It should at least be a list, better a named tuple or object, so you can put names to the entries and not have to refer to the entries by some magic number.